oddt/scoring/__init__.py

```diff
@@ -127,32 +129,78 @@ def _gen_pdbbind_desc(self,
             n_jobs = -1
         else:
             n_jobs = self.n_jobs
+
+        blacklist = []
+        if use_proteins:
+            # list of protein files that segfault OB 2.4.1
+            blacklist = ['1sqq', '1rbo', '1e8h', '1sqp', '2wij',
+                         '1sqb', '2wig', '3axk', '2f2h', '2wik',
+                         '1nu1', '1ntk', '3cf1', '3axm']
+
+        desc_idx = [pid for pid in df.index.values
+                    if pdbbind_db[pid].pocket is not None]
         result = Parallel(n_jobs=n_jobs, verbose=1)(
             delayed(method_caller)(
                 self.descriptor_generator,
                 'build',
                 [pdbbind_db[pid].ligand],
-                protein=pdbbind_db[pid].pocket)
-            for pid in df.index.values if pdbbind_db[pid].pocket is not None)
-        descs = np.vstack(result)
-        for i in range(len(self.descriptor_generator)):
-            df[str(i)] = descs[:, i]
-        df.to_csv(desc_path, float_format='%.5g')
+                protein=getattr(pdbbind_db[pid], 'protein' if use_proteins
+                                else 'pocket'))
+            for pid in desc_idx if pid not in blacklist)
+
+        # sparse descs may have different shapes, dense are stored np.array
+        if sparse:
+            descs = result
+        else:
+            descs = np.vstack(result)
+
+        # create dataframe with descriptors with pdbids as index
+        df_desc = pd.DataFrame(descs, index=desc_idx)
+        df_desc.index.rename('pdbid', inplace=True)
+
+        # for sparse features leave one column and cast explicitly to list
+        if sparse:
+            df_desc.columns = ['sparse']
+            df_desc['sparse'] = df_desc['sparse'].map(lambda x: list(x.flatten()))
+
+        # DF are joined by index (pdbid) since some might be missing
+        df.join(df_desc, how='inner').to_csv(desc_path, float_format='%.5g')
 
-    def _load_pdbbind_desc(self, desc_path, pdbbind_version=2016):
+    def _load_pdbbind_desc(self, desc_path, pdbbind_version=2016,
+                           train_set='refined', test_set='core'):
 
         df = pd.read_csv(desc_path, index_col='pdbid')
 
-        train_set = 'refined'
-        test_set = 'core'
+        # generate dense representation of sparse descriptor in CSV
         cols = list(map(str, range(len(self.descriptor_generator))))
-        train_idx = (df['%i_%s' % (pdbbind_version, train_set)] &
-                     ~df['%i_%s' % (pdbbind_version, test_set)])
-        self.train_descs = df.loc[train_idx, cols].values
+        if 'sparse' in df.columns:
```

Comment:
Is it possible to have a column called `sparse` and some other columns in this data frame?

Reply:
Not one generated by `scorer` class